### PR TITLE
Knaster-Tarski for categories

### PR DESCRIPTION
**Pull request by TOTBWF**
# Description

This PR proves a Knaster-Tarski theorem for categories, and uses it to prove the usual Knaster-Tarski theorem for suplattices.

## Checklist

Before submitting a merge request, please check the items below:

- [x] I've read [the contributing guidelines](https://github.com/plt-amy/1lab/blob/main/CONTRIBUTING.md).
- [x] The imports of new modules have been sorted with `support/sort-imports.hs` (or `nix run --experimental-features nix-command -f . sort-imports`).
- [x] All new code blocks have "agda" as their language.

If your change affects many files without adding substantial content, and
you don't want your name to appear on those pages (for example, treewide
refactorings or reformattings), start the commit message and PR title with `chore:`.

### CODE REVIEW

**Comment by Lavenza** [comment]
[Pull request preview](https://preview.1lab.dev/436)

File: src/Cat/Functor/Algebra/Limits.lagda.md
Comment (ncfavier) [inline src/Cat/Functor/Algebra/Limits.lagda.md:1]: Can this be combined with `Cat.Diagram.Monad.Limits` somehow (even theoretically)?

File: src/Cat/Functor/Algebra/KnasterTarski.lagda.md [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+
+This problem of overly ambitious limits is similar to the one encountered
+in the naïve [[adjoint functor theorem]], so we can use a similar technique
+to repair our proof. In particular, we will impose a variant of the
+**solution set condition** on the category of $F$-algebras that ensures
+that the limit we end up computing is determined by a small amount of data,
+which lets us relax our large-completeness requirement.
+
+Explicilty, we will require that the category of $F$-algebras has a
+small [[weakly initial family]] of algebras. Explicitly, we require:
```
Comment (ncfavier) [inline src/Cat/Functor/Algebra/KnasterTarski.lagda.md:52]:
```suggestion
Explicitly, we will require that the category of $F$-algebras has a
small [[weakly initial family]] of algebras. Explicitly, we require:
```

Also, 'explicitly' twice.

File: src/Cat/Functor/Algebra/Limits.lagda.md
Comment (TOTBWF) [reply src/Cat/Functor/Algebra/Limits.lagda.md:1]: There should be some general theory of reflection/creation of limits in displayed categories that covers both cases

Comment (TOTBWF) [reply src/Cat/Functor/Algebra/Limits.lagda.md:1]: I've done some poking around at this, and we can tell quite a nice story in terms of liftings of kan extensions! This does require quite a bit of general theory though, so it's probably best left to a separate PR.

**Review by ncfavier — APPROVED** [review]

**Pull request merged**